core/controllers/topic_editor_test.py

```diff
@@ -0,0 +1,199 @@
+# limitations under the License.
+
+"""Tests for the topic editor page."""
+
+from core.domain import story_services
+from core.domain import topic_services
+from core.domain import user_services
+from core.tests import test_utils
+import feconf
+
+
+class BaseTopicEditorControllerTest(test_utils.GenericTestBase):
+
+    def setUp(self):
+        """Completes the sign-up process for the various users."""
+        super(BaseTopicEditorControllerTest, self).setUp()
+        self.signup(self.TOPIC_MANAGER_EMAIL, self.TOPIC_MANAGER_USERNAME)
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.topic_manager_id = self.get_user_id_from_email(
+            self.TOPIC_MANAGER_EMAIL)
+        self.new_user_id = self.get_user_id_from_email(
+            self.NEW_USER_EMAIL)
+
+        self.set_admins([self.ADMIN_USERNAME])
+        self.set_topic_managers([self.TOPIC_MANAGER_USERNAME])
+
+        self.topic_manager = user_services.UserActionsInfo(
+            self.topic_manager_id)
+        self.admin = user_services.UserActionsInfo(self.admin_id)
+        self.new_user = user_services.UserActionsInfo(self.new_user_id)
+        self.topic_id = topic_services.get_new_topic_id()
+        self.save_new_topic(
+            self.topic_id, self.admin_id, 'Name', 'Description', [], [], [])
+
+
+class NewStoryHandlerTest(BaseTopicEditorControllerTest):
+
+    def test_story_creation(self):
+        self.login(self.ADMIN_EMAIL)
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            response = self.testapp.get(
+                '%s/%s' % (feconf.TOPIC_EDITOR_URL_PREFIX, self.topic_id))
+            csrf_token = self.get_csrf_token_from_response(response)
+            json_response = self.post_json(
+                '%s/%s' % (feconf.NEW_STORY_URL, self.topic_id),
+                {'title': 'Story title'},
+                csrf_token=csrf_token)
+            topic = topic_services.get_topic_by_id(self.topic_id)
+            story_id = json_response['storyId']
+            self.assertEqual(len(story_id), 12)
+            self.assertEqual(topic.canonical_story_ids, [story_id])
+            self.assertIsNotNone(
+                story_services.get_story_by_id(story_id, strict=False))
+        self.logout()
+
+
+class TopicEditorTest(BaseTopicEditorControllerTest):
+
+    def test_access_topic_editor_page(self):
+        """Test access to editor pages for the sample topic."""
+
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            # Check that non-admin and topic_manager cannot access the editor
+            # page.
+            response = self.testapp.get(
+                '%s/%s' % (feconf.TOPIC_EDITOR_URL_PREFIX, self.topic_id))
+            self.assertEqual(response.status_int, 302)
+
+            # Check that topic admins can access and edit in the editor
```

Comment:
topic admins --> admins. Might want to check separately for topic manager as well.

Reply:
Done.